src/sql/workbench/contrib/connection/browser/connectionStatus.ts

```diff
@@ -59,16 +64,38 @@ export class ConnectionStatusbarItem extends Disposable implements IWorkbenchCon
 	// Update the connection status shown in the bar
 	private _updateStatus(): void {
 		let activeConnection = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
+		let id = undefined;
+		if (this._editorService.activeEditor) {
+			// USE ACTIVE EDITOR INFO AS THE ID WILL BE DIFFERENT FOR EDITOR CONNECTION.
+			let newInfo = this._connectionManagementService.getConnectionInfo(this._editorService.activeEditor.resource.toString());
+			id = newInfo?.serverConnectionId;
```

Comment:
With the current logic it's possible to have the global connection be different than the editor connection if OE is currently selected, which means we could end up displaying connection info from one connection and the SPID from a completely different connection. We should probably fix it so that never happens - although exactly how the behavior works when an OE connection is selected is something that should be discussed.

Reply:
Fixed it so that the URI is checked with the current editor and active connection to see if they match first.